src/net/sourceforge/kolmafia/KoLCharacter.java

```diff
@@ -3905,47 +3905,25 @@ private static void addAvailableSkill(final UseSkillRequest skill, final boolean
 
   /** Adds a single skill to the list of skills temporarily possessed by this character. */
   public static final void addAvailableCombatSkill(final int skillId) {
-    KoLCharacter.addAvailableCombatSkill(UseSkillRequest.getUnmodifiedInstance(skillId));
+    if (!KoLConstants.availableCombatSkillsSet.contains(skillId)) {
+      KoLConstants.availableCombatSkillsSet.add(skillId);
+    }
   }
 
   public static final void addAvailableCombatSkill(final String skillName) {
     // *** Skills can have ambiguous names. Best to use the methods that deal with skill id
-    KoLCharacter.addAvailableCombatSkill(UseSkillRequest.getUnmodifiedInstance(skillName));
-  }
-
-  private static void addAvailableCombatSkill(final UseSkillRequest skill) {
-    if (skill == null) {
-      return;
-    }
-
-    if (KoLConstants.availableCombatSkillsMap.containsKey(skill)) {
-      return;
-    }
-
-    KoLConstants.availableCombatSkills.add(skill);
-    KoLConstants.availableCombatSkillsMap.put(skill, null);
+    KoLCharacter.addAvailableCombatSkill(SkillDatabase.getSkillId(skillName));
   }
 
   public static final void removeAvailableCombatSkill(final int skillId) {
-    KoLCharacter.removeAvailableCombatSkill(UseSkillRequest.getUnmodifiedInstance(skillId));
+    if (KoLConstants.availableCombatSkillsSet.contains(skillId)) {
```

Comment:
I think you can skip the contains: if you call `remove`, and the item isn't present, the function returns false (which you can just ignore).

Reply:
Done.